### DISCUSSION TITLE
Middleware and encryption

### DISCUSSION THREAD

**Discussion by Extarys** [geldata/gel#6698] [Q&A]
I'm trying EdgeDB and with Prisma I was using `prisma-field-encryption` to avoid storing PII in plain text.

I believe EdgeDb doesn't (yet?) support middleware? Is there a suggested way of approaching this, and are middlewares on the roadmap? :)

EDIT: Talking about security, a middleware or some schema option could allow us to never return a user's hashed password unless explicitly asked for.

Thanks!

**Comment by elprans** [comment]
We should be able to expose basic encryption capabilities via the [pgcrypto](https://www.edgedb.com/docs/stdlib/pgcrypto) extension.
